### PR TITLE
DSS-5543: Fixes rendering of datasets in detailview navigating from s…

### PR DESCRIPTION
**Pull request by theseyi**
…idebar/ treeview.

@ericsun2 this replaces the last PR

### CODE REVIEW

**Comment by ericsun2** [comment]
LGTM
Thanks for the fix

File: web/public/javascripts/routers/datasets.js [hunk 1]
```diff
@@ -208,32 +208,39 @@ App.DatasetRoute = Ember.Route.extend({
...
+            type: firstType === 'null' ? type : [firstType, ...type]
+          };
+        });
 
-      // Flatten nested structure if present
-      return [].concat(...getFieldTypeSet(JSON.parse(schema))); // TODO: cover n-th dimension, if expected
+        // Flatten nested structure if present
+        return [].concat(...getFieldTypeSet(JSON.parse(schema))); // TODO: cover n-th dimension, if expected
+      };
+      Promise.all([Ember.$.getJSON(`api/v1/datasets/${id}`), Ember.$.getJSON(`api/v1/datasets/${id}/security`)])
```
Comment (luisah) [inline web/public/javascripts/routers/datasets.js]: I suggest using [Ember.rsvp.all](http://emberjs.com/api/classes/RSVP.html#method_all)

Comment (theseyi) [inline web/public/javascripts/routers/datasets.js]: Hmm... rather than native Promises?

Comment (luisah) [inline web/public/javascripts/routers/datasets.js]: I know Ember prefers using Ember methods instead of any solution but it's your call!

Comment (theseyi) [inline web/public/javascripts/routers/datasets.js]: Makes sense, i'll update with that, thanks for the feedback

**Pull request merged**